cpp/arcticdb/version/version_store_api.cpp

```diff
@@ -1122,4 +1125,11 @@ void PythonVersionStore::force_delete_symbol(const StreamId& stream_id) {
     delete_all_for_stream(store(), stream_id, true);
 }
 
+std::lock_guard<std::mutex> PythonVersionStore::ensure_single_thread_cpp_pybind_entry(){ 
+    py::gil_scoped_release release;
```

Comment:
Why is there a gil release call here? IMO this does a bit more than the name implies.

Reply:
Can you also add this information as a comment above the function (preferably in doxygen style)?